### PR TITLE
Eager loading of snapshot properties

### PR DESCRIPTION
**Pull request by ahadas**
Previously, with lazy loading, snapshot properties could have been dropped in the following flow:
1. Update a snapshot configuration
2. Quit the application so the snapshot configuration would be persisted to snapshot.xml
3. Start the application
4. Close the application
5. Start the application
6. The snapshot configuration was reverted to its default value

The reason why the snapshot configuration was removed is because it wasn't loaded between step 3 and step 4 and therefore when closing the application, we cleared the snapshot configuration that is going to be persisted (in order to get rid of properties that were dropped) and then, when updating the snapshot configuration, we didn't get updated value of the snapshot configuration from step 1.

The solution is to override the `read` method of `MuSnapshotable` to make sure the snapshot configuration are loaded on startup.

### CODE REVIEW

File: mucommander-core/src/main/java/com/mucommander/search/SearchSnapshot.java [hunk 1]
```diff
@@ -16,11 +16,24 @@
...
 import com.mucommander.commons.conf.Configuration;
 import com.mucommander.snapshot.MuSnapshot;
 import com.mucommander.snapshot.MuSnapshotable;
 
 public class SearchSnapshot implements MuSnapshotable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SearchSnapshot.class);
+
+    @Override
+    public void read(Configuration configuration) {
+        LOGGER.info("Loading snapshot configuration for " + SearchSnapshot.class);
```
Comment (pskowronek) [inline mucommander-core/src/main/java/com/mucommander/search/SearchSnapshot.java:31]: LOGGER.info("Loading snapshot configuration for: {}" , SearchSnapshot.class); ?

File: mucommander-core/src/main/java/com/mucommander/ui/viewer/EditorSnapshot.java [hunk 2]
```diff
@@ -16,18 +16,39 @@
...
 
 /**
  * Snapshot preferences for editor.
  */
 public final class EditorSnapshot implements MuSnapshotable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EditorSnapshot.class);
+
+    @Override
+    public void read(Configuration configuration) {
+        LOGGER.info("Loading snapshot configuration for " + EditorSnapshot.class);
```
Comment (pskowronek) [inline mucommander-core/src/main/java/com/mucommander/ui/viewer/EditorSnapshot.java:36]: ....... {}

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerSnapshot.java [hunk 3]
```diff
@@ -25,12 +31,32 @@
...
 public final class TextViewerSnapshot implements MuSnapshotable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TextViewerSnapshot.class);
+
+    /**
+     * Section describing information specific to text file presenter.
+     */
+    public static final String TEXT_FILE_PRESENTER_SECTION = FILE_PRESENTER_SECTION + "." + "text";
+
+    @Override
+    public void read(Configuration configuration) {
```
Comment (pskowronek) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerSnapshot.java:42]: A lot of dup'd code - why not move it to some base class?

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerPreferences.java [hunk 4]
```diff
@@ -78,27 +100,35 @@ enum EditorViewerMode {
...
-        this.value = defaultValue;
-        this.mode = mode;
+
+        Metadata metadata;
+        try {
+            metadata = getDeclaringClass().getDeclaredField(this.name()).getDeclaredAnnotation(Metadata.class);
+            mode = metadata.mode();
+            value = metadata.value();
+        } catch (NoSuchFieldException | SecurityException e) {
+            // ignore, the properties are initialized with the annotation's default values
```
Comment (pskowronek) [inline mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerPreferences.java:130]: LOGGER.trace ?

**Review by pskowronek — COMMENTED** [review]
pls see my comments

**Comment by ahadas** [comment]
thanks @pskowronek, please take another look

**Review by pskowronek — APPROVED** [review]
looks good

**Pull request merged**